src/main/scala/za/co/absa/commons/lang/extensions/OptionExtension.scala

```diff
@@ -32,6 +32,15 @@ object OptionExtension {
     def toTry(failure: => Exception): Try[T] =
       option.map(Success(_)).getOrElse(Failure(failure))
 
+    /**
+     * Gets the `option` value or throws the provided exception
+     *
+     * @param exception the exception to throw in case the `option` is None
+     * @return
+     */
+    def getOrThrow(exception: => Throwable): T = {
+      option.getOrElse(throw exception)
+    }
```

Comment:
`getOrThrow(new exception)` seems to be very similar to calling `option.getOrElse(throw new exception)` is there some reason why `getOrElse` is not enough?

Reply:
:100: Just posted the same question to the issue? :)

Reply:
Well, `Option.getOrElse` does work of course.
`getOrThrow` is few characters shorter, and explicitly shows what will happen if the `Option` is `None`.
(Do you question the coexistence of `filter` and `filterNot`? 😉 )

Reply:
Well... first of all, `filter` and `filterNot` are both in a standard library, so I cannot really question them :)
But even if I could I'd probably wouldn't because `filterNot` is useful for DSL. When is called with a predicate function, it gives you a cleaner notation that you cannot achieve with `filter` + `!` combination. Consider this:
```scala
xs filter isEven       // clean
xs filterNot isEven    // clean
xs filter (!isEven(_)) // doh! :(
```

Reply:
Don't know how to settle this other than by vote 😉 
Add your emoticons:
👍  It's a great function, I really would welcome it in _Commons_
👀  The function is OK, but not really worth to be part of _Commons_
👎  Nah, it's just garbage. In complicates the code - any code, unnecessary

Reply:
The function is fine, but it is just too similar to standard solution. And in that case it's usually better to use the already existing function.

By the way I had already some commons contribution implemented to later realise it's actually possible to use some standard way to do the same thing, So I understand where you are coming from.

Reply:
> to later realise it's actually possible to use some standard way

:100: %, been there as well :)

Reply:
OK, scraping here.
But you might stumble at the function in some libs still (meaning I won't be removing it from already existing routines) 😉